File: tests/basic/label_remove.ml
open Optitrust
open Target

let _ = Run.doc_script_cpp (fun _ ->
    !! Label_basic.remove [cLabel "mylabel"]
  )
"
int main() {
  int a = 0;
  mylabel: int b = 1;
  int c = 2;
}
"

let _ = Run.script_cpp ( fun _ ->

      !! Label_basic.remove [cLabel "start"];
      !! Label_basic.remove [cLabel "unit"];
      !! List.iter (fun l -> Label_basic.remove [cLabel l]) ["loop";"cond";"incr_1";"incr_2"];
      !! Tools.failure_expected (fun () ->
            Label_basic.remove [cLabel "foo"];)
)

File: tests/basic/label_remove.cpp
#include <stdbool.h>

int main() {

  int x;
  start:
   x = 3;

  loop: for (int i = 0; i < 3; i++) {

  cond: if (1) {
    incr_1: x++;
  } else {
    incr_2: x--;
    }
  }

   stop: return 0;
  }

